### PR TITLE
[ticket/16129] Enhance attachment's ALT tag

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16129

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16129

### CODE REVIEW

**Comment by 3D-I** [comment]
!set Styles

File: phpBB/styles/prosilver/template/attachment.html [hunk 1]
```diff
@@ -8,14 +8,14 @@
 
 		<!-- IF _file.S_THUMBNAIL -->
 		<dl class="thumbnail">
-			<dt><a href="{_file.U_DOWNLOAD_LINK}"><img src="{_file.THUMB_IMAGE}" class="postimage" alt="{_file.DOWNLOAD_NAME}" title="{_file.DOWNLOAD_NAME} ({_file.FILESIZE} {_file.SIZE_LANG}) {_file.L_DOWNLOAD_COUNT}" /></a></dt>
+			<dt><a href="{_file.U_DOWNLOAD_LINK}"><img src="{_file.THUMB_IMAGE}" class="postimage" alt="{% if _file.COMMENT %}{{ _file.COMMENT }}{% else %}{{ _file.DOWNLOAD_NAME }}{% endif %}" title="{_file.DOWNLOAD_NAME} ({_file.FILESIZE} {_file.SIZE_LANG}) {_file.L_DOWNLOAD_COUNT}" /></a></dt>
```
Comment (VSEphpbb) [inline phpBB/styles/prosilver/template/attachment.html]: Alternatively you can use ternary syntax:
`{{ _file.COMMENT ?: _file.DOWNLOAD_NAME }}`

Comment (mrgoldy) [reply phpBB/styles/prosilver/template/attachment.html]: I believe Hanakin is not too fond of that though.

Comment (DavidIQ) [reply phpBB/styles/prosilver/template/attachment.html]: What happens if there is a quote in the name or comment? Isn't there something else needed here for escaping?

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: @DavidIQ I will test it and amend the code if necessary.

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: @VSEphpbb Yes but the main reason for this is code's readability, in any case let's see what @hanakin says about that.

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: @DavidIQ it is ok.

immagine

Comment (DavidIQ) [reply phpBB/styles/prosilver/template/attachment.html]: That just looks like it's rendering out the quote inside of the alt property making the markup invalid.

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: Hmm, I commented with 

immagine

Comment (VSEphpbb) [reply phpBB/styles/prosilver/template/attachment.html]: Needs to use an escape, like `_file.COMMENT|e('html')`

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: Just on the comment side?

Comment (VSEphpbb) [reply phpBB/styles/prosilver/template/attachment.html]: yes. File download name's shouldn't have html chars in them

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: How the screen reader will "read" this?

immagine

Comment (VSEphpbb) [reply phpBB/styles/prosilver/template/attachment.html]: That's proper HTML markup now.

Comment (3D-I) [reply phpBB/styles/prosilver/template/attachment.html]: Ok

**Comment by 3D-I** [comment]
Thx guys. :+1:

File: phpBB/styles/prosilver/template/attachment.html (hunk 1 above)
Comment (hanakin) [reply phpBB/styles/prosilver/template/attachment.html]: Please no ternary in template files

**Comment by marc1706** [comment]
While the escaped characters might not "sound" great when using a screenreader, I think they're fine. In essence one can only hope that users will set a proper comment for the files they upload.

**Pull request merged**